libs/parser/parser.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Parser for config files
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include "parser.h"
+
+#define MAX_HEADER_LEN 16
+#define MAX_FIELD_LEN  16
+#define MAX_VALUE_LEN  64
+
+#define WORD "[A-Za-z0-9_.,-]"
+
+
+/*
+ * Patterns for regex expressions
+ *
+ * If you have changed this pattern check if these function work properly:
+ * - parser_nextHeaderGet
+ * - parser_nextFieldNameGet
+ * - parser_nextFieldGet
+ *
+ * Especially be aware of `regmatch_t` structures.
+ */
+#define HEADER_PATTERN        "^[[:space:]]*@(" WORD "+)[[:space:]]*(#.*)?[[:space:]]*$"
+#define HEADER_SUBEXPRESSIONS 2
+
+#define FIELD_PATTERN        "^[[:space:]]*(" WORD "+)[ =]+(" WORD "+)[[:space:]]*(#.*)?[[:space:]]*$"
+#define FIELD_SUBEXPRESSIONS 3
+
+#define COMMENT_PATTERN "^[[:space:]]*#.*$|^[[:space:]]*$"
+
+/* Structure contains information about defined header */
+typedef struct {
+	char headerName[MAX_HEADER_LEN + 1];
+	int fieldsNo;
+	int (*converter)(hmap_t *);
+} parser_headerInfo_t;
+
+/* Structure of the main parser object */
+struct parser_t {
+	parser_headerInfo_t *headerInfos;
+	hmap_t *headersInfosMap;
+
+	int maxHeadersNb;
+	int defHeaders;
+	int maxFieldsNb;
+};
+
+typedef struct {
+	char field[MAX_FIELD_LEN + 1];
+	char value[MAX_VALUE_LEN + 1];
+} parser_field_t;
+
+
+static int instancesNb = 0;
+
+static regex_t headerRegex;
+static regex_t fieldRegex;
+static regex_t commentRegex;
```

Comment:
Please use _common_ structure for such variables.
```suggestion
struct {
	int instancesNb;
	regex_t headerRegex;
	regex_t fieldRegex;
	regex_t commentRegex;
} parser_common;
```